ER図を作成 ToDo App.drawio

Ryuji-Hayashi35 opened the pull request.

Ryuji-Hayashi35 committed:
ER図を作成 ToDo App.drawio

busitora commented:
- 概要欄に、ER図のスクリーンショット及びリンクを添えて下さい！
- 概要欄に、各カラムの説明(一般的な分かるものは除く)を追記して下さい！

Ryuji-Hayashi35 committed:
Update ToDo App.drawio

スクショのリンクとurl、カラムの説明を追加しました。

Tsuchiya2 commented:
@Ryuji-Hayashi35 
ここでいう概要欄とは↓の赤枠のことですね！
レビュワーがすぐに確認できるようにPRを作成することを意識してPR修正してみましょう！
[![Image from Gyazo](https://t.gyazo.com/teams/startup-technology/0998f7dc482c6d7e703c64324220ccbc.png)](https://startup-technology.gyazo.com/0998f7dc482c6d7e703c64324220ccbc)

Ryuji-Hayashi35 committed:
Update ToDo App.drawio

すみません、勘違いしていました。
概要欄にER図のスクショとurl、カラムの説明を追加しました。

ER図作成
https://app.diagrams.net/#HRyuji-Hayashi35%2Fgoal_setting_app%2Fcreate_readme%2FToDo%20App.drawio

[![Image from Gyazo](https://i.gyazo.com/8ee9c205ee1ecdc1be8bab154ecb8cfe.png)](https://gyazo.com/8ee9c205ee1ecdc1be8bab154ecb8cfe)

User
  name
  password
  email
  created_at
  updated_at

Task
  user_id
  title
  scheduled＿days(完了までの日数)
  memo
  created_at
  updated_at

Good(いいね機能)
  user_id 
  task_id
  created_at
  updated_at

Role
  role
  created_at
  updated_at

Ryuji-Hayashi35 commented:
すみません、勘違いしていました。
概要欄にER図のスクショとurl、カラムの説明を追加しました。

ER図作成
https://app.diagrams.net/#HRyuji-Hayashi35%2Fgoal_setting_app%2Fcreate_readme%2FToDo%20App.drawio

[![Image from Gyazo](https://i.gyazo.com/8ee9c205ee1ecdc1be8bab154ecb8cfe.png)](https://gyazo.com/8ee9c205ee1ecdc1be8bab154ecb8cfe)

User
  name
  password
  email
  created_at
  updated_at

Task
  user_id
  title
  scheduled＿days(完了までの日数)
  memo
  created_at
  updated_at

Good(いいね機能)
  user_id 
  task_id
  created_at
  updated_at

Role
  role
  created_at
  updated_at

Tsuchiya2 commented:
以下の部分が気になりました。
- 目標を公開・非公開に設定できる用のカラムは必要ないか確認してみると良いかも。
- ログイン機能にはsorceryを使うならカラムが不足しているかも。
- カラムmemoは何を保存するものなのかが気になりました。
- 目標の達成・未達成が分かるカラムは必要ないか確認してみると良いかも。
- サボった日数はどこで検知するのでしょうか？
- 今のER図で管理ユーザーと一般ユーザーを区別できるようになっているか確認してみると良いかも。

上記よく再考して再度提出をお願いします！

Ryuji-Hayashi35 commented:
ご指摘いただいた点を以下のように修正しました。
・目標を公開、非公開に設定できるようstatusカラムを追加しました。
・ログイン機能に必要なsaltカラムを追加しました。
・カラムmemoは目標の詳細、具体的に何をするかを記述する場所として想定しています。
・目標の達成、未達成がわかるカラムとしてtask_completed,task_failedカラムを追加しました。
・サボった日数を検知するcount_daysカラムを追加しました。
・Userテーブルに一般ユーザーと管理ユーザーを区別するroleカラムを追加しました。

ER図作成
https://app.diagrams.net/#L名称未設定ファイル.drawio

[![Image from Gyazo](https://i.gyazo.com/96f63bbbfc282433b13e0c1642cb1def.png)](https://gyazo.com/96f63bbbfc282433b13e0c1642cb1def)

User
name
password
email
salt
created_at
updated_at
role

Task
user_id
title
scheduled＿days(完了までの日数)
memo(タスクで具体的に何をするかメモ)
count_days(サボった日数カウント)
status(公開非公開設定)
task_completed(タスク達成)
task_failed(タスク未達成)
created_at
updated_at

Good(いいね機能)
user_id
task_id
created_at
updated_at

Role
role
created_at
updated_at

Tsuchiya2 commented:
ありがとうございます！
以下コメントになります。

---

- Userテーブルのpasswordカラムはsorcery使用時はこのカラム名で良かったか確認の必要がありそうです。
- Userテーブルにroleカラムがある状態で、Roleテーブルは何を担うテーブルになるか確認してみる必要がありそうです。
- Taskテーブルのmemoカラムも「タスクの内容」という意味であれば他の言葉の方が個人的には適切かと感じました。（memoだとちょっとした走り書きみたいなイメージを抱いたため）
- task_completed, task_failedカラムのところですが、本当に2つ必要そうですかね？boolean型で真偽を問うカラムで良さそうな気がしました。

Ryuji-Hayashi35 commented:
ご指摘いただいた点を以下のように修正しました。
・Userテーブルのpasswordカラムをcrypted_passwordに変更しました。
・Taskテーブルのmemoカラムをtextカラムに変更しました。
・task_completed, task_failedカラムをbooleanカラムに変更しました。
roleテーブルに関しては、まだよくわからないのでヒントをいただけると幸いです。

ER図作成
https://app.diagrams.net/#L名称未設定ファイル.drawio

[![Image from Gyazo](https://i.gyazo.com/c4cb935e9de081959f7c51e8c947c9d4.png)](https://gyazo.com/c4cb935e9de081959f7c51e8c947c9d4)

User
name
crypted_password
email
salt
created_at
updated_at
role

Task
user_id
title
scheduled＿days(完了までの日数)
text(タスクで具体的に何をするかメモ)
count_days(サボった日数カウント)
status(公開非公開設定)
boolean(タスク達成、未達成判別)
created_at
updated_at

Good(いいね機能)
user_id
task_id
created_at
updated_at

Role
role
created_at
updated_at

Tsuchiya2 commented:
ありがとうございます！
以下コメントになります。

---

- boolean型：これはstring型, text型, integer型とかの1つですね！なのでカラム名はachievement(達成)とかの単語の方が良いかと！
- Userテーブルにroleカラムがありますね！こちらで一般ユーザーなのか、管理ユーザーなのかを分けることができるかと思うのですが、そうなるとRoleテーブルは何をするテーブルになりますかね？既にUserテーブル内で一般・管理ユーザーを分けられるようになっている状況で、このRoleテーブルは必要なのかどうか再考してみましょう！

Ryuji-Hayashi35 commented:
ありがとうございます！
ご指摘いただいた点を以下のように修正しました。
・booleanカラムをachievementに変更しました。
・roleテーブルを削除しました。

ER図作成
https://app.diagrams.net/#L名称未設定ファイル.drawio

[![Image from Gyazo](https://i.gyazo.com/9f88e09e7144cabd1b6ec9d2ffe55f7b.png)](https://gyazo.com/9f88e09e7144cabd1b6ec9d2ffe55f7b)

User
name
crypted_password
email
salt
created_at
updated_at
role

Task
user_id
title
scheduled＿days(完了までの日数)
text(タスクで具体的に何をするかメモ)
count_days(サボった日数カウント)
status(公開非公開設定)
achievement(タスク達成、未達成判別)
created_at
updated_at

Good(いいね機能)
user_id
task_id
created_at
updated_at

Tsuchiya2 commented:
@Ryuji-Hayashi35 
LGTMです！
テーブル数が少ないので、就活に入る前までに追加機能を考えて実装していきましょう！（目安としては7つ以上あると良いかと）

Ryuji-Hayashi35 commented:
ありがとうございます！
